File: examples/collections/kotlin/core/checked.kt
package salvo.core.checked

data class Checked<T>(
    val value: T,
)

fun<T> checked(value: T): Checked<T> {
    return Checked(value = value)
}

fun<T> ignore(checked: Checked<T>) {
    (checked).let {}
}

fun<T> detach(checked: Checked<T>): T {
    return checked.value
}

File: examples/collections/rust/core/checked.rs
use crate::core_iterator::*;

#[derive(Clone, Debug, PartialEq)]
pub struct Checked<T: Clone + 'static> {
    pub value: T,
}

pub fn checked<T: Clone>(value: T) -> Checked<T> {
    return Checked { value: value };
}

pub fn ignore<T: Clone>(checked: Checked<T>) {
    drop(checked);
}

pub fn detach<T: Clone>(checked: Checked<T>) -> T {
    return checked.value;
}
